src/SpecialCase/ComposerJsonZendFrameworkPackageSpecialCase.php

```diff
@@ -0,0 +1,670 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-migration for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-migration/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-migration/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\Migration\SpecialCase;
+
+use Composer\Semver\Semver;
+use RuntimeException;
+
+class ComposerJsonZendFrameworkPackageSpecialCase implements SpecialCaseInterface
+{
+    public function matches($filename, $content)
+    {
+        if (! preg_match('#\bcomposer\.json$#', $filename)) {
+            return false;
+        }
+
+        $composer = json_decode($content, true);
+        return isset($composer['require']['zendframework/zendframework'])
+            || isset($composer['require-dev']['zendframework/zendframework']);
+    }
+
+    public function replace($content)
+    {
+        $composer = json_decode($content, true);
+
+        $isProduction = isset($composer['require']['zendframework/zendframework']);
```

Comment:
I'd suggest small refactor here:
```php
$section = isset($composer['require']['zendframework/zendframework'])
    ? 'require'
    : 'require-dev';

$packages = $this->getPackageList($composer[$section]['zendframework/zendframework']);
$composer = $this->updateComposer($composer, $packages, $section);
```

so then we just have one ternary operation.